FillableFieldworkAIDriver.lua

```diff
@@ -57,86 +62,117 @@ end
 
 --- Drive the refill part of the course
 function FillableFieldworkAIDriver:driveUnloadOrRefill()
+	if self:getSiloSelectedFillTypeSetting():isEmpty() then 
+		self:setSpeed(0)
+		self:setInfoText('NO_SELECTED_FILLTYPE')
+		return
+	else
+		self:clearInfoText('NO_SELECTED_FILLTYPE')
+	end
 	local isNearWaitPoint, waitPointIx = self.course:hasWaitPointWithinDistance(self.ppc:getCurrentWaypointIx(), 5)
-
-	self:searchForRefillTriggers()
+	self.waitPointIx = waitPointIx
+	local isInWaitPointRange = self.waitPointIx and self.waitPointIx+8 >self.ppc:getCurrentWaypointIx() or isNearWaitPoint 
+	if self:is_a(FieldSupplyAIDriver) then
+		if not isInWaitPointRange  then
+			courseplay:isTriggerAvailable(self.vehicle)
```

Comment:
isTriggerAvaliable is a misleading name it suggests this is just a check returning a boolean with no side effects. In reality this looks more like a activateTriggerWhenAvailable() or something similar.